snapcraft/tests/fake_servers.py

```diff
@@ -235,14 +235,18 @@ def do_POST(self):
 
     def _handle_upload_request(self):
         logger.info('Handling upload request')
-        self.send_response(200)
-        self.send_header('Content-Type', 'application/octet-stream')
+        if 'UPDOWN_BROKEN' in os.environ:
```

Comment:
Why do this via environment variables instead of setting something on the fixture?